src/main.rs

```diff
@@ -41,43 +33,100 @@ fn main() -> Result<()> {
         _default_app = conf.default_app;
     }
 
-    match verb {
-        Verbs::create => {
-            let data = util::json_parse(sub_cmd.unwrap().value_of(Parameters::data))?;
-            match resource {
-                Resources::app => apps::create(&url, id, data)?,
-                Resources::device => {
-                    let app_id = sub_cmd.unwrap().value_of(Resources::app).unwrap();
-                    devices::create(&url, id, data, app_id)?
-                },
-            }
-        }
-        Verbs::delete => {
-            match resource {
-                Resources::app => apps::delete(&url, id)?,
-                Resources::device => {
-                    let app_id = sub_cmd.unwrap().value_of(Resources::app).unwrap();
-                    devices::delete(&url, app_id, id)?
-                },
-            }
-        }
-        Verbs::edit => {
-            match resource {
-                Resources::app => apps::edit(&url, id),
-                Resources::device => {
-                    let app_id = sub_cmd.unwrap().value_of(Resources::app).unwrap();
-                    devices::edit(&url, app_id, id)
-                },
-            }
-        }
-        Verbs::get => {
-            match resource {
-                Resources::app => apps::read(&url, id)?,
-                Resources::device => {
-                    let app_id = sub_cmd.unwrap().value_of(Resources::app).unwrap();
-                    devices::read(&url, app_id, id)?
-                },
+    match matches.subcommand() {
+        
+        (cmd_name, _cmd) => {
```

Comment:
As `_cmd` is used, you should remove the leading `_`. It's a convention to show that you ignore the value which is not the case here.